### PR TITLE
API review perf issue when loading with documentation tags

### PR DESCRIPTION
**Pull request by praveenkuttappan**
Tag entire line as documentation if all tokens are within doc range. Instead of tagging each token as documentation start a span of documentation around tokens.

Instead of hiding documentation after loading, set them as hidden by default.

### CODE REVIEW

File: src/dotnet/APIView/APIView/CodeFileRenderer.cs [hunk 1]
```diff
@@ -24,23 +25,40 @@ private void Render(List<CodeLine> list, IEnumerable<CodeFileToken> node)
...
+            bool isLineAllDocumentation = true;
 
             foreach (var token in node)
             {
                 switch(token.Kind)
                 {
                     case CodeFileTokenKind.Newline:
-                        list.Add(new CodeLine(stringBuilder.ToString(), currentId));
+                        //Close documentation span if within doc range
+                        if(isDocumentation)
```
Comment (pakrym) [inline src/dotnet/APIView/APIView/CodeFileRenderer.cs:38]: Formatting, here and elsewhere.
```suggestion
                        if (isDocumentation)
```

File: src/dotnet/APIView/APIView/CodeLine.cs [hunk 2]
```diff
@@ -6,10 +6,13 @@ namespace ApiView
     {
         public string DisplayString { get; }
         public string ElementId { get; }
-        public CodeLine(string html, string id)
+
+        public bool isLineAllDocumentation { get; }
```
Comment (pakrym) [inline src/dotnet/APIView/APIView/CodeLine.cs:10]: Formatting and naming
```suggestion
        public bool IsDocumentationLine { get; }

```

File: src/dotnet/APIView/APIViewWeb/Pages/Assemblies/_CodeLine.cshtml [hunk 3]
```diff
@@ -9,9 +9,11 @@
         DiffLineKind.Added => "code-added",
         _ => ""
     };
+    string documentationClass = Model.CodeLine.isLineAllDocumentation? "documentation": "";
+
```
Comment (pakrym) [inline src/dotnet/APIView/APIViewWeb/Pages/Assemblies/_CodeLine.cshtml:13]: nit: extra nl

File: src/dotnet/APIView/APIView/CodeFileRenderer.cs [hunk 4]
```diff
@@ -10,7 +10,8 @@ namespace ApiView
     public class CodeFileRenderer
     {
         public static CodeFileRenderer Instance = new CodeFileRenderer();
-
+        const string DOCUMENTATION_SPAN_START = "<span class=\"documentation\">";
```
Comment (pakrym) [inline src/dotnet/APIView/APIView/CodeFileRenderer.cs:13]:
```suggestion
        private const string DOCUMENTATION_SPAN_START = "<span class=\"documentation\">";
```

And we typically put consts before other members.

File: src/dotnet/APIView/APIView/CodeFileRenderer.cs [hunk 5]
```diff
@@ -24,23 +25,40 @@ private void Render(List<CodeLine> list, IEnumerable<CodeFileToken> node)
             string currentId = null;
             bool isDocumentation = false;
             bool isDeprecatedToken = false;
+            //This will be set to true by default when a new line starts and 
+            // set to false when any non documentation token is found within the line
+            bool isLineAllDocumentation = true;
```
Comment (pakrym) [inline src/dotnet/APIView/APIView/CodeFileRenderer.cs:30]: What about empty lines?

Comment (praveenkuttappan) [reply src/dotnet/APIView/APIView/CodeFileRenderer.cs:30]: correct. I should set this as false and set it true when range started and if line is empty. Similarly this flag should not be blindly reset to tru when a new line starts and it should be set if it is only within range. Thanks for the suggestions.

**Review by pakrym — APPROVED** [review]

**Pull request merged**